peer/brontide.go

```diff
@@ -1660,6 +1660,29 @@ func (p *Brontide) handleCustomMessage(msg *lnwire.Custom) error {
 	return p.cfg.HandleCustomMessage(p.PubKey(), msg)
 }
 
+// isLoadedFromDisk returns true if the provided channel ID is loaded from
+// disk.
+func (p *Brontide) isLoadedFromDisk(chanID lnwire.ChannelID) bool {
+	p.activeChanMtx.RLock()
+	defer p.activeChanMtx.RUnlock()
+
+	// If this is a newly added channel, no need to reestablish.
+	_, added := p.addedChannels[chanID]
+	if added {
+		return false
+	}
```

Comment:
previously we use `nil` value to indicate it's a pending channel. And because we only ever have pending channels in the `activeChannels` map during startup, it also implicitly means it's loaded from disk. With the following commits this implication is not correct anymore.

On the other hand, the loaded channels won't appear in `addedChannels` map, so we perform an explicit check here.